packages/api/AlvTime.Persistence/Repositories/AlvEconomyData/OvertimePayoutStorage.cs

```diff
@@ -0,0 +1,117 @@
+    {
+        private readonly AlvEconomyDataContext _economyContext;
+        private readonly IEmployeeHourlySalaryStorage _employeeHourlySalaryStorage;
+
+        public OvertimePayoutStorage(AlvEconomyDataContext economyContext, IEmployeeHourlySalaryStorage employeeHourlySalaryStorage)
+        {
+            _economyContext = economyContext;
+            _employeeHourlySalaryStorage = employeeHourlySalaryStorage;
+        }
+
+        public OvertimePayoutResponsDto DeleteOvertimePayout(int userId, DateTime date)
+        {
+            var overtimePayout =
+                _economyContext.OvertimePayouts.FirstOrDefault(op => op.UserId == userId && op.Date.Equals(date));
+
+            if (overtimePayout == null)
+            {
+                throw new ValidationException("Could not find a payout registered for this user");
+            }
+
+            _economyContext.OvertimePayouts.Remove(overtimePayout);
+            _economyContext.SaveChanges();
+
+            return new OvertimePayoutResponsDto
+            {
+                
+                Id = overtimePayout.Id,
+                Date = overtimePayout.Date,
+                UserId = overtimePayout.UserId,
+                TotalPayout = overtimePayout.TotalPayout
+            };
+        }
+
+        public void SaveTotalOvertimePayout(RegisterOvertimePayoutDto overtimePayout)
+        {
+            _economyContext.OvertimePayouts.Add(new OvertimePayout
+            {
+                UserId = overtimePayout.UserId,
+                Date = overtimePayout.Date,
+                TotalPayout = overtimePayout.TotalPayout
+            });
+            _economyContext.SaveChanges();
+        }
+
+        public decimal RegisterOvertimePayoutSalary(List<OvertimeEntry> overtimeEntries, int userId, GenericHourEntry requestedPayout)
+        {
+            var overtimeEntriesForPayout = GetOvertimeEntriesForPayout(overtimeEntries, requestedPayout.Hours);
+            var overtimeSalary = GetOvertimeSalaryPayout(overtimeEntriesForPayout, userId);
+
+            SaveTotalOvertimePayout(new RegisterOvertimePayoutDto
+            {
+                TotalPayout = overtimeSalary,
+                UserId = userId,
+                Date = requestedPayout.Date
+            });
+
+            return overtimeSalary;
+        }
+
+        private decimal GetOvertimeSalaryPayout(List<OvertimeEntry> overtimeEntriesForPayout, int userId)
+        {
+            var salary = 0.0M;
+
+            foreach (var overtimeEntry in overtimeEntriesForPayout)
+            {
+                var hourlySalary = _employeeHourlySalaryStorage.GetHouerlySalary(userId, overtimeEntry.Date);
+
+                salary += hourlySalary * overtimeEntry.Hours * overtimeEntry.CompensationRate;
+            }
+
+            return salary;
+        }
+        private List<OvertimeEntry> GetOvertimeEntriesForPayout(List<OvertimeEntry> overtimeEntries, decimal hoursForPayout)
```

Comment:
Klarer ikke helt å følge denne metoden hele veien gjennom. Kanskje du kan forklare meg den på Alvtime-torsdag? :)

Reply:
Ja, veldig gjerne, for det burde være en bedre/ penere måte å gjøre dette på, men var den eneste jeg kom på som faktisk ga riktige resultater